src/pages/Login/Login.js

```
@@ -1,7 +1,42 @@
-import { Link } from "react-router-dom";
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useState } from "react";
 import { useDocumentTitle } from "../../hooks/useDocumentTitle";
-import "./Login.css"
+import "./Login.css";
+import axios from "axios";
+import { useAuth } from "../../contexts";
 export const Login = () => {
+  const [user, setUser] = useState({ email: "", password: "" });
+  const navigate = useNavigate();
+  const location = useLocation();
+  const { authDispatch } = useAuth();
+
+  const [hidePass, sethidePass] = useState(true);
+
+  const showHide = () => {
+    hidePass ? sethidePass(false) : sethidePass(true);
+  };
+
+  const loginButtonHandler = async (e, userData) => {
+    e.preventDefault();
+    try {
+      const response = await axios.post("/api/auth/login", userData);
+      const { foundUser: user, encodedToken } = response.data;
+      if (response.status === 200) {
+        authDispatch({
+          type: "AUTH_SUCCESS",
+          payload: { user, encodedToken, isAuth: true },
+        });
+        localStorage.setItem("user", JSON.stringify(user));
+        localStorage.setItem("encodedToken", encodedToken);
+        localStorage.setItem("isAuth", true);
+        navigate(location.state?.from?.pathname || "/", { replace: true });
+      }
+    } catch (error) {
+      console.log(error);
```

Instead of logging error, you can use react toastify package to display toast for login success/failure or wrong password

---

yup i will be doing this in next pr